### ISSUE TITLE
UnusedPrivateProperty does not detect unused property named "logger"

### ISSUE THREAD

**Issue by Chris-Goetz** [detekt/detekt#7222]
## Expected Behavior
detekt should find unused properties for which no exception is defined. 

## Observed Behavior
detekt does not find unused properties named "logger".

## Steps to Reproduce
![Screenshot from 2024-04-24 13-29-18](https://github.com/detekt/detekt/assets/141140288/80cdd1f6-098f-49c9-bc76-46f5f2865297)

## Context
The documentation statesthe default allowed values are: allowedNames (default: '_|ignored|expected|serialVersionUID') - "logger" is not listed here.
We also have no exceptions in our config for that rule defined:
```
---
build:
  weights:
    LongParameterList: 0
    TooManyFunctions: 0

complexity:
  excludes: ["**/test/**"]

formatting:
  MaximumLineLength:
    excludes: ["**/test/**"]
  NoWildcardImports:
    active: false
  TrailingCommaOnCallSite:
    active: true
  TrailingCommaOnDeclarationSite:
    active: true

naming:
  FunctionParameterNaming:
    excludes: ["**/test/**"]
  ClassNaming:
    excludes: ["**/test/**"]
  MemberNameEqualsClassName:
    active: false

style:
  MaxLineLength:
    excludes: ["**/test/**"]
  WildcardImport:
    active: false
  SerialVersionUIDInSerializableClass:
    active: false

config:
  validation: true

```

## Your Environment
* Version of detekt used: 1.23.5
* Version of Gradle used (if applicable):
------------------------------------------------------------
Gradle 8.6
------------------------------------------------------------
Build time:   2024-02-02 16:47:16 UTC
Revision:     d55c486870a0dc6f6278f53d21381396d0741c6e

Kotlin:       1.9.20
Groovy:       3.0.17
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          17.0.10 (Private Build 17.0.10+7-Ubuntu-122.04.1)
OS:           Linux 6.5.0-28-generic amd64

**Comment by Chris-Goetz** [comment]
Hi,
sorry for the late answer, one of my Outlook filters has eaten up the notification.

I've tried to run 'detektMain' wich gave me a lot of new findings but the unused logger property was still ignored.
Then I tried to assign the value '2' to logger, like in your test. That was found by both, the 'detekt' and the 'detektMain' task.
After that, I tried to assing 'private val logger = Logger.getLogger("foo")' and that was also found.

It must have something to do with the 'KotlinLogging.logger {}'.

**Comment by atulgpt** [comment]
Hi @Chris-Goetz I tried below TCs

```kotlin
@Test
fun `#7222 1`() {
    val code = """
        interface KLogger
        object KotlinLogging {
            fun logger(func: () -> Unit): KLogger = TODO()
        }

        private val logger = KotlinLogging.logger {}
    """.trimIndent()
    assertThat(subject.lintWithContext(env, code)).hasSize(1)
}

@Test
fun `#7222 2`() {
    val code = """
        interface KLogger
        object KotlinLogging {
            fun logger(func: () -> Unit): KLogger = TODO()
        }

        private val logger = KotlinLogging.logger {}
        private val something = KotlinLogging.logger {}
    """.trimIndent()
    assertThat(subject.lintWithContext(env, code)).hasSize(2)
}
```

and it is still running fine.

Also rule `UnusedPrivateProperty` required type resolution which means it will only work correctly when `detektMain` is used(NOT `detekt`). 
Can you see from the TC is something is still missing here?
